packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -2019,13 +2024,21 @@ class _CupertinoTimerPickerState extends State<CupertinoTimerPicker> {
   void dispose() {
     PaintingBinding.instance.systemFonts.removeListener(_handleSystemFontsChange);
     textPainter.dispose();
+
+    _hourScrollController?.dispose();
+    _minuteScrollController?.dispose();
+    _secondScrollController?.dispose();
     super.dispose();
   }
 
   @override
   void didUpdateWidget(CupertinoTimerPicker oldWidget) {
     super.didUpdateWidget(oldWidget);
 
+    _hourScrollController?.jumpToItem(selectedHour!);
+    _minuteScrollController?.jumpToItem(selectedMinute ~/ widget.minuteInterval);
+    _secondScrollController?.jumpToItem(selectedSecond! ~/ widget.secondInterval);
+
     assert(
```

Comment:
The selected values change in number of cases: when widget is update and when user clicks different item. 
We need to cover all of them. How about invoking `jumpToItem` on every build? Then we do not need to do it on didUpdateWidget.

Reply:
To my knowledge `jumpToItem` can't be invoked during build, as it triggers a rebuild. The way I see it this would require a refactor of hour/minute/second pickers into their own widgets.

But there is another thing I noticed while trying to add a test for the correct item being centered that is that I can't notice a difference in widget behavior. The only instance where they have an effect is during the first build with an `initialTimerDuration` being set. 
This leads me to believe that the recreation with `initialItem` is just an unnecessary quirk of the old code.

Reply:
Good place to invoke jumpToItem is event handlers that change selected values. Then no refactoring is required.

Reply:
Thank you for pushing this PR forward!

Reply:
> But there is another thing I noticed while trying to add a test for the correct item being centered that is that I can't notice a difference in widget behavior. 

How scroll position is changed when user change selection then? May be an experiment in an application may help.

Reply:
Here is a simplified example with the internal widgets where `initialItem` has no effect beyond the first build:
```dart
class A extends StatefulWidget {
  const A({super.key});

  @override
  State<A> createState() => _AState();
}

class _AState extends State<A> {
  int selected = 10;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        const SizedBox(height: 80,),
        Text('Selected: $selected'),
        CupertinoPicker(
            scrollController: FixedExtentScrollController(
              initialItem: 10
            ),
            itemExtent: 50,
            onSelectedItemChanged: (int idx) {
              setState(() {
                selected = idx;
              });
            },
            children: [
              for(int i = 0; i<25; i++)
                Text(i.toString())
            ]
        ),
      ],
    );
  }
}
```

Here is a even simpler example with the base classes, that also won't jump to the start when rebuilding:
```dart
SizedBox(
  height: 200,
  child: ListView.builder(
      itemCount: 100,
      controller: ScrollController(
          initialScrollOffset: 200
      ),
      itemBuilder: (context, idx) => Text('Item $idx')
  ),
)
```

The reason a rebuild didn't change the scroll position was that the scroll offset was restored from context.

Scroll controllers save position whenever scrolling is finished:
https://github.com/flutter/flutter/blob/29b25165cab883d290f6245ec267550bf6058cd9/packages/flutter/lib/src/widgets/scroll_position.dart#L549
Loaded as part of state restoration the old position gets loaded, if it is not null (so not at the first build): https://github.com/flutter/flutter/blob/29b25165cab883d290f6245ec267550bf6058cd9/packages/flutter/lib/src/widgets/scrollable.dart#L617

So if it is okay with you I would remove the `jumpToItem` entirely, as it would actually differ from the original behavior.

Reply:
Interesting. Thank you for investigation.
But main question is: is there bug in the widget or the position is changed as expected? If there is a bug let's create an issue.
If original initialPosition did not take effect, yes, it is ok to remove jumpToItem.

Reply:
The changes LGTM!